x/incentives/keeper/gauge_test.go

```diff
@@ -332,3 +332,80 @@ func (suite *KeeperTestSuite) TestChargeFeeIfSufficientFeeDenomBalance() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestAddToGaugeRewards() {
+	testCases := []struct {
+		name               string
+		owner              sdk.AccAddress
+		coinsToAdd         sdk.Coins
+		gaugeId            uint64
+		minimumGasConsumed uint64
+
+		expectErr bool
+	}{
+		{
+			name:  "valid case: valid gauge",
+			owner: suite.TestAccs[0],
+			coinsToAdd: sdk.NewCoins(
+				sdk.NewCoin("uosmo", sdk.NewInt(100000)),
+				sdk.NewCoin("atom", sdk.NewInt(99999)),
+			),
+			gaugeId:            1,
+			minimumGasConsumed: uint64(2 * types.BaseGasFeeForAddRewardToGauge),
```

Comment:
Since the base gas cost for txns like these are usually in the 10k-20k gas range, we should add a test case with many denoms (>4) to ensure that the `minimumGasConsumed` check isn't passing trivially

Reply:
sounds good added! for 8 tokens it looks like this; 

```
gasConsumed: 161441  tc. minimumGasConsumed: 80000
```